frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionListItem.tsx

```diff
@@ -44,77 +46,73 @@ function ImplicitActionCardContent() {
 
 function ModelActionListItem({
   action,
-  editorUrl,
+  actionUrl,
   canWrite,
   onArchive,
 }: Props) {
-  const isImplicitAction = action.type === "implicit";
-
+  const canArchive = canWrite && action.type !== "implicit";
   const { show: askConfirmation, modalContent: confirmationModal } =
     useConfirmation();
 
   const handleArchive = useCallback(() => {
-    if (!onArchive) {
-      return;
-    }
     askConfirmation({
       title: t`Archive ${action.name}?`,
       confirmButtonText: t`Archive`,
-      onConfirm: onArchive,
+      onConfirm: () => onArchive(action),
     });
   }, [action, askConfirmation, onArchive]);
 
-  const renderCardContent = () =>
-    action.type === "query" ? (
-      <QueryActionCardContent action={action} />
-    ) : action.type === "implicit" ? (
-      <ImplicitActionCardContent />
-    ) : null;
-
-  const canArchive = onArchive && !isImplicitAction;
-
-  const menuItems = [
-    canArchive && {
-      title: t`Archive`,
-      icon: "archive",
-      action: handleArchive,
-    },
-  ].filter(Boolean);
-
-  const subtitleParts = [
-    action.public_uuid && t`Public Action`,
-    // Remove this optional chaining after removed all the existing actions without creators
-    action.creator?.common_name && t`Created by ${action.creator.common_name}`,
-  ].filter(isNotNull);
-
-  const hasMenu = menuItems.length > 0;
+  const menuItems = useMemo(
+    () => [
+      {
+        title: canWrite ? t`Edit` : t`View`,
+        icon: canWrite ? `pencil` : `eye`,
```

Comment:
nit: let's not use backticks when we don't need a template string

Reply:
Fixed